### DISCUSSION TITLE
Whats the Best-Practice when extending system collections?

### DISCUSSION THREAD

**Discussion by DanielBiegler** [directus/directus#10141] [Archive]
Hey, I was wondering if theres a consensus on how to go about extending the system collections.

For example I'd be interested in tracking when a user got registered, i.e. a `date_creation` field, but by default theres none. (I'm up to date on `v9.1.2`) - It's not just meta info, I'd imagine there are use cases where you want to add fields like an address or domain specific fields.

So we could extend the system collection itself which is cool but couldnt this hurt us in the long run when migrating or whatever?

I tried it out and added a field to `Directus Users` and users are by default forbidden to update said field which is nice.

We also could make a new collection `Custom User Info` with a unique, non-nullable, cascading many-to-one relation from `"Directus Users".id` to a `"Custom User Info".id_user` field. Now we have a collection thats not hidden in the system dropdown with its entirely own permissions.

What do y'all think, how do you go about your users or other system collections?

Cheers.

**Comment by benhaynes** [comment]
There's no issue with adding your own custom fields to the system collections. Though, it is a good point that it brings our proprietary tables into your schema. In the future, the plan is to allows using your own User/File tables for managing that info... but to do that we need a to way to map _your_ User/File fields to the system fields the App/API requires/expects.

**Reply by DanielBiegler** [reply]
> it is a good point that it brings our proprietary tables into your schema

Yeah I think this is the crux of it followed by an issue. I was just messing around with a sqlite setup but you can't delete added fields to the `Directus Users` Collection because sqlite doesnt handle dropping a column like one would expect. It rebuilds it and drops the whole table which is **not possible** due to foreign key constraints. Sqlite users beware! :warning: 

So now I'll will go down the second route with a `Custom User Info`-Collection to be portable.

**Reply by rijkvanzanten** [reply]
> Collection because sqlite doesnt handle dropping a column like one would expect. It rebuilds it and drops the whole table which is not possible due to foreign key constraints. Sqlite users beware! 

This was recently fixed in https://github.com/knex/knex/pull/4189 👍🏻
